packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -447,6 +508,14 @@ export function commitUpdate(
   // Update the props handle so that we know which props are the ones with
   // with current event handlers.
   updateFiberProps(domElement, newProps);
+
+  switch (type) {
+    case 'img': {
+      if (oldProps.src !== newProps.src) {
```

Comment:
Hmm... what if we delayed the setting of src to commitMount instead of finalizeInitialChildren? Then the deduping macrotask would work on mount too and if we ended up unmounting before commit finished we'd have no event listeners still in the tree so the event would noop